src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs

```diff
@@ -6,6 +6,8 @@ namespace System.Globalization
 {
     internal static partial class GlobalizationMode
     {
+        internal static bool UseIcu => true;
```

Comment:
```suggestion
        internal const bool UseIcu = true;
```

This will allow C# compiler to strip the Icu code.

Reply:
I believe it is also stripped as it is, but the change makes sense.

Reply:
I tried making a similar change a while back for an unrelated code path and it ended up causing a bunch of "Unreachable code detected" build warnings. So be prepared to scatter a bunch of pragmas throughout the managed globalization call paths if we go this route.

Reply:
Thanks. I'll try changing it and see how it goes 🤞

Reply:
It will produce unreachable warnings

Reply:
It would be unfortunate to ship Windows-specific stuff on Unix. It would either disable the warnings (e.g. even for the whole file); or resort to ifdefs.

Reply:
Once we get https://github.com/dotnet/runtime/issues/31785 , it can be a property and the linker can dead-code eliminate the Windows-specific code on Unix. So having an imperfect workaround - like disabling the warnings for the whole file - may be acceptable for now.

Reply:
#31785 is actually not required for this to work as the property is constant in the implementation. The linker will do the dead-code elimination correctly today

Reply:
We do not have this enabled when building CoreLib today. What needs to happen to enable it? Can we do this before this change?

Reply:
I think it's if SPC is processed by linker during build (it's default setting on linker side)

Reply:
I did this test before putting the PR and it stripped the Windows implementation on Unix correctly. I'll do another smoke test today to make sure and share screenshots and more detailed information but I believe @marek-safar is correct here.

Reply:
I just validated this and I can see that it is actually being trimmed correctly. Here is the result from the pre-trim output: 
![image](https://user-images.githubusercontent.com/22899328/78832379-00995b00-79a0-11ea-9886-28add736d811.png)
![image](https://user-images.githubusercontent.com/22899328/78832425-14dd5800-79a0-11ea-9c2b-30a339041e6a.png)

Here is the result of searching for Nls on ILSpy on the final System.Private.CoreLib which is running the ILTrimmer:
![image](https://user-images.githubusercontent.com/22899328/78832159-a4ced200-799f-11ea-8155-8775000782c7.png)

![image](https://user-images.githubusercontent.com/22899328/78832228-bc0dbf80-799f-11ea-8938-52ab55d25812.png)

It might be worth adding a test into System.Runtime.Tests that tries and gets some of the windows types and methods to validate the trimmer is trimming and protect from any regression there. Thoughts?

Are we OK with leaving as a property?

Reply:
> Are we OK with leaving as a property?

Sounds good.

I got confused by the calls to `Globalization.UseIcu` still being there. I expected them to disappear as well. @marek-safar Is there a reason for not stripping them?

Reply:
Yes, it's a linker and the goal was to remove dependencies and not do inlining which has significantly more complex rules, breaks debugging, etc. We could add another pass for limited inlining cases but I'm not sure it's worth it as existing JIT/AOT engines can handle this.

Reply:
@marek-safar it seems that if I change `UseIcu` in Linux to be `UseIcu { get; } = GetUseIcuMode();` and then `GetUseIcuMode() => true;` the trimmer doesn't strip out any code, is that a known limitation?

Reply:
That is expected. It is introducing non-trivial data flow that the linker is not able to crack.